packages/fields/src/types/Content/views/Controller.js

```diff
@@ -62,7 +68,39 @@ export default class ContentController extends TextController {
       // Forcibly return null if empty string
       return { document: null };
     }
-    return { document: JSON.stringify(data[path].document) };
+
+    let blocks;
+    // May return synchronously, or may throw with either an actual error or a
+    // loading promise. We should never see a Promise thrown as .serialize()
+    // only gets called during event handlers on the client _after_ all the
+    // React Suspense calls are fully resolved. We want the
+    // returns-synchronously case. Otherwise, we want to either rethrow any
+    // error thrown, or throw a new error indicating an unexpected Promise was
+    // thrown.
+    try {
+      blocks = this.getBlocks();
+    } catch (loadingPromiseOrError) {
+      if (isPromise(loadingPromiseOrError)) {
+        // `.getBlocks()` thinks it's in React Suspense mode, which we can't
+        // handle here, so we throw a new error.
+        throw new Error(
+          '`Content#getBlocks()` threw a Promise. This may occur when calling `Content#serialize()` before blocks have had a chance to fully load.'
```

Comment:
Is this a case that can actually happen right now or is it to provide a nice error for someone in the future if they change something which makes this be called at a different time?

Reply:
For future safety. While I was hacking on this code I could easily see a future where refactoring accidentally called this at some point before blocks were loaded, so wanted to make that possible failure case explicit.